galsim/gsobject.py

```diff
@@ -1713,6 +1721,9 @@ def drawPhot(self, image, n_photons=0, rng=None, max_extra_noise=None, poisson_f
 
             phot_array.scaleFlux(g * thisN / Ntot)
 
+            for op in surface_ops:
```

Comment:
Question: I think that the way you've set this up (as a loop over a bunch of things to be done in order using an ```applyTo``` method) makes sense.  But what I'm wondering about is information flow.  Will some of these processes need access to additional information besides what is flowing through here?  Can we assume that if necessary the PhotonArray class will be modified to carry more info around, or might this need to have some more complex way of passing through the relevant info?  I realize I'm asking you to speculate about potentially arbitrarily complicated sensor effects so possibly the answer is just "I don't know, we'll see if it happens" which I guess is also legit.